waiter/src/waiter/scheduler/shell.clj

```
@@ -291,6 +311,36 @@
       (deliver completion-promise :failed)
       id->service)))
 
+(defn- signal-instance-fn
+  "Deletes the instance corresponding to service-id/instance-id and returns if successful"
+  [id->service service-id instance-id message port->reservation-atom port-grace-period-ms signal-type]
+  (try
+    (if (contains? id->service service-id)
+      (let [{:keys [id->instance]} (get id->service service-id)
+            {:keys [:shell-scheduler/process] :as instance} (get id->instance instance-id)]
+        (if (and instance (active? instance))
+          (do
+            (log/info "signaling instance" instance-id "process" process "with signal" signal-type)
+            (case signal-type
+              :sigkill (kill-process! instance port->reservation-atom port-grace-period-ms)
+              :sigterm (safe-kill-process! instance port->reservation-atom port-grace-period-ms))
```

Please throw error for an unsupported signal type.